addons/delivery/models/stock_picking.py

```diff
@@ -115,23 +115,28 @@ def put_in_pack(self):
         package_id = super(StockPicking, self).put_in_pack()
         package = self.env['stock.quant.package'].browse(package_id)
 
-        # By default, sum the weights of all package operations contained in this package
-        pack_operation_ids = self.env['stock.pack.operation'].search([('result_package_id', '=', package_id)])
-        package_weight = sum([x.qty_done * x.product_id.weight for x in pack_operation_ids])
-        package.shipping_weight = package_weight
-
-        return {
-            'name': _('Package Details'),
-            'type': 'ir.actions.act_window',
-            'view_mode': 'form',
-            'res_model': 'stock.quant.package',
-            'view_id': self.env.ref('delivery.view_quant_package_form_save').id,
-            'target': 'new',
-            'res_id': package_id,
-            'context': {
-                'current_package_carrier_type': self.carrier_id.delivery_type if self.carrier_id.delivery_type not in ['base_on_rule', 'fixed'] else 'none',
-            },
-        }
+        current_package_carrier_type = self.carrier_id.delivery_type if self.carrier_id.delivery_type not in ['base_on_rule', 'fixed'] else 'none'
+        count_packaging = int(self.env['product.packaging'].search_count([('package_carrier_type', '=', current_package_carrier_type)]))
```

Comment:
* Why convert the result of search_count to int? Isn't it already one?
* If there is no need to open a wizard, why doe this return the package_id?
* Also I assume `count_packaging` can't be negative, so you can probably just check if it's truthy or falsy
* Final suggestion: just insert a check with an early return, that way there's no need to reindent half the function:
  ```python
  if not count_packaging:
      return ???

  # generate the package details action as before
  ```